homeassistant/components/synology_dsm/backup.py

```
@@ -161,15 +162,13 @@ async def async_delete_backup(
 
         :param backup_id: The ID of the backup that was returned in async_list_backups.
         """
-        try:
+        with suppress(SynologyDSMAPIErrorException):
             await self._file_station.delete_file(
                 path=self.path, filename=f"{backup_id}.tar"
             )
             await self._file_station.delete_file(
                 path=self.path, filename=f"{backup_id}_meta.json"
             )
-        except SynologyDSMAPIErrorException as err:
```

do we have a more granular exception here? because this sounds  like you would suppress any error, not only `NotFound`

---

there is no more specific exception yet. we might check the error code, but we didn't do this before, so i didn't add it now. Further the whole exception logic needs to be reworked in the lib, but this is something for the future 🙈

---

can you check if the error code is possible? Just suppressing everything without even logging anything is not a good idea imo

---

the exception model really needs to be overhauled in the lib :see_no_evil: